tests/common/test_logcollector.py

```diff
@@ -0,0 +1,440 @@
+        must_collect_files = [
+            os.path.join(self.root_collect_dir, "waagent*"),
+            os.path.join(self.root_collect_dir, "less_important_file*")
+        ]
+
+        with patch("azurelinuxagent.common.logcollector.ARCHIVE_SIZE_LIMIT", 10 * 1024 * 1024):
+            with patch("azurelinuxagent.common.logcollector.MUST_COLLECT_FILES", must_collect_files):
+                lc = LogCollector(self.manifest_path)
+                archive = lc.collect_logs()
+
+        expected_files = [
+            os.path.join(self.root_collect_dir, "waagent.log"),
+            os.path.join(self.root_collect_dir, "waagent.log.1"),
+            os.path.join(self.root_collect_dir, "waagent.log.2.gz")
+        ]
+        unexpected_files = [
+            os.path.join(self.root_collect_dir, "waagent.log.3.gz"),
+            os.path.join(self.root_collect_dir, "less_important_file"),
+            os.path.join(self.root_collect_dir, "another_dir", "least_important_file")
+        ]
+        self._assert_files_are_in_archive(expected_files)
+        self._assert_files_are_not_in_archive(unexpected_files)
+
+        self._assert_archive_created(archive)
+
+        no_files = self._get_number_of_files_in_archive()
+        self.assertEquals(3, no_files, "Expected 3 files in archive, found {0}!".format(no_files))
+
+        # Second collection, if a file got deleted, delete it from the archive and add next file on the priority list
+        # if there is enough space.
+        rm_files(os.path.join(self.root_collect_dir, "waagent.log.3.gz"))
+
+        with patch("azurelinuxagent.common.logcollector.ARCHIVE_SIZE_LIMIT", 10 * 1024 * 1024):
+            with patch("azurelinuxagent.common.logcollector.MUST_COLLECT_FILES", must_collect_files):
+                second_archive = lc.collect_logs()
+
+        expected_files = [
+            os.path.join(self.root_collect_dir, "waagent.log"),
+            os.path.join(self.root_collect_dir, "waagent.log.1"),
+            os.path.join(self.root_collect_dir, "waagent.log.2.gz"),
+            os.path.join(self.root_collect_dir, "less_important_file"),
+            os.path.join(self.root_collect_dir, "another_dir", "least_important_file")
+        ]
+        unexpected_files = [
+            os.path.join(self.root_collect_dir, "waagent.log.3.gz")
+        ]
+        self._assert_files_are_in_archive(expected_files)
+        self._assert_files_are_not_in_archive(unexpected_files)
+
+        self._assert_archive_created(second_archive)
+
+        no_files = self._get_number_of_files_in_archive()
+        self.assertEquals(5, no_files, "Expected 5 files in archive, found {0}!".format(no_files))
+        self._assert_archive_sizes()
+
+    def test_log_collector_should_update_archive_when_files_are_new_or_modified_or_deleted(self):
+        # Ensure the archive reflects the state of files on the disk at collection time. If a file was updated, it
```

Comment:
seems to me like the archive is an internal implementation detail... should this test be done in terms of the zip?

Reply:
Forgot to resolve, archive step is gone so the tests assert on the zip anyway.